source/user_guide.rst

```diff
@@ -13,148 +14,465 @@ Who should read this
+        "1"
+      ],
+      "label": "device"
+    }'
+
+
+The template ID list on line 6 contains the only template ID configured so far.
+To check out the configured device, just send a GET request to /device:
+
+.. code-block:: bash
+
+    curl -X GET http://localhost:8000/device -H "Authorization: Bearer ${JWT}"
+
+
+Which should give back:
 
-Remember that the token must be set in the request header as a whole, not parts of it. 
-In the example only the first characters are shown for the sake of simplicity.
+.. code-block:: bash
+
+    {
+      "pagination": {
+        "has_next": false,
+        "next_page": null,
+        "total": 1,
+        "page": 1
+      },
+      "devices": [
+        {
+          "templates": [
+            1
+          ],
+          "created": "2018-01-25T12:36:29.353958+00:00",
+          "attrs": {
+            "1": [
+              {
+                "template_id": "1",
+                "created": "2018-01-25T12:30:42.167126+00:00",
+                "label": "temperature",
+                "value_type": "float",
+                "type": "dynamic",
+                "id": 1
+              }
+            ]
+          },
+          "id": "0998",
+          "label": "device_0"
+        }
+      ]
+    }
+
+
+Sending messages
+++++++++++++++++
+
+So far we got an access token and created a template and a device based on it.
+In an actual deploy, the physical device would send messages to dojot with all
+its attributes and their current values. For this tutorial we will send MQTT
+messages by hand to the platform, emulating such physical device. For that, we
+will use mosquitto_pub from Mosquitto project.
+
+.. ATTENTION::
+    Some Linux distributions, Ubuntu in particular, have two packages for
+    `mosquitto`_ - one containing tools to access it (such mosquitto_pub and
+    mosquitto_sub for publishing messages and subscribing to topics) and
+    another one containing the MQTT broker. In this tutorial, only the tools
+    are going to be used. There are some reports of problems that were caused
+    because the broker was mistakenly installed and automatically started,
+    which prevents the MQTT broker instance from dojot's docker-compose to be
+    initiated (and therefore used). If you are facing problems such as "I send
+    messages via MQTT but nothing happens", please first check if you are not
+    running an extra instance of mosquitto in your machine outside docker.
+
+
+The dojot compatible format for messages sent by devices is a simple key-value 
+JSON, such as:
+
+.. code-block:: json
+
+    {
+      "temperature" : 10.6
+    }
+
+Let's send this message to dojot:
+
+.. code-block:: bash
+
+  mosquitto_pub -t /admin/0998/attrs -m '{"temperature": 10.6}'
+
+If there is no output, the message was sent to MQTT broker. The topic is build 
+from the following information:
+
+- admin: user tenant. This is retrieved from user "service" attribute (used
+  when getting the access token).
```

Comment:
actually, this is when the user is created isn't it?

Reply:
Yes, fixed.